### PR TITLE
feat(datepicker-popover): adds support of mountNode

### PR DESCRIPTION
**Pull request by jcemer**
#### Description

It adds support of mountNode attribute to Datepicker and Popover. It creates feature compatibility with Modal.

#### Scope

- [ ] Patch: Bug Fix
- [x] Minor: New Feature
- [ ] Major: Breaking Change

### CODE REVIEW

File: src/popover/popover.js [hunk 1]
```diff
@@ -399,6 +399,16 @@ class Popover extends React.Component<PopoverPropsT, PopoverPrivateStateT> {
...
   }
 
+  getMountNode(): HTMLElement {
+    const {mountNode} = this.props;
+    if (mountNode) {
+      return mountNode;
+    }
+    // Flow thinks body could be null (cast through any)
+    // eslint-disable-next-line flowtype/no-weak-types
+    return ((document.body: any): HTMLBodyElement);
```
Comment (nadiia) [inline src/popover/popover.js]: Need to wrap it in the `if (__BROWSER__) {}`

**Comment by schnerd** [comment]
Seems like this makes sense – probably something that may need to be supported in any component that uses portal.

Would anyone prefer a `getMountNode: () => HTMLElement` callback instead of a `mountNode: HTMLElement` prop? This is something I didn't fully explore when building `Modal` so I settled on the latter, but I'm wondering if anyone sees benefits to using the callback style:

```js
getPopoverMountNode() {
  return document.querySelector('.fake-body');
}

render() {
  return <Popover getMountNode={this.getPopoverMountNode} ...>
}
```
vs
```
getPopoverMountNode() {
  return document.querySelector('.fake-body');
}

render() {
  const mountNode = document.querySelector('.fake-body');
  return <Popover mountNode={mountNode} ...>;
}
```

If they're essentially the same, then we can roll with whats in the current PR.

**Comment by gergelyke** [comment]
@jcemer would you mind updating the jest snapshots / tests?

File: src/popover/popover.js (hunk 1 above)
Comment (jcemer) [reply src/popover/popover.js]: I'm not sure because I get this implementation from the Modal component and I can't see the condition there.

Comment (jcemer) [reply src/popover/popover.js]: It is already checked on `render`.

**Comment by jcemer** [comment]
@nadiia @gergelyke I fixed the tests.

**Comment by gergelyke** [comment]
thanks @jcemer ! would you mind signing our CLA ? https://cla-assistant.io/uber-web/baseui?pullRequest=901

**Comment by jcemer** [comment]
@gergelyke done.

**Comment by tajo** [comment]
What is the use-case of using custom mountNode for any of those components? SSR?

**Comment by jcemer** [comment]
@tajo It is for the cases there is a need to attach the popover in another element than document.body. 

In our case, we are rendering inside an iframe using a portal. The popover is mounting in the parent window so there is a need to have `mountNode` to set it to the document.body of the iframe. The modal already allows us to define it.

**Comment by jcemer** [comment]
@nadiia @chasestarr @gergelyke I updated the branch, let me know if there is something missing. It would be great to have it since datepicker is stable now 🎉

**Comment by gergelyke** [comment]
thanks, @jcemer! as Nadiia recommended, can you please add test coverage?

**Comment by jcemer** [comment]
@gergelyke @nadiia could you advise how to write it because I couldn't find a test that covers it in the modal component.

**Comment by nadiia** [comment]
Let's merge this change as is.

**Comment by gergelyke** [comment]
@jcemer would you mind resolving this (hopefully) last merge conflict? We'll land it after that's done

**Comment by gergelyke** [comment]
and thanks for your patience! :)

**Comment by jcemer** [comment]
@gergelyke done :)

**Comment by chasestarr** [comment]
take a look at these docs for running the e2e tests: https://github.com/uber-web/baseui/blob/master/TESTING.md

**Comment by nadiia** [comment]
Merging. Thanks again for the contribution!

**Pull request merged**